src/hooks/useENS.ts

```diff
@@ -1,23 +1,41 @@
 import { useEffect, useState } from "react"
+import { ChainId } from "../constants"
 import { ethers } from "ethers"
 import { useActiveWeb3React } from "."
 
 type ReturnType = { ensName: string | null }
 
 export const useENS = (address: string | null | undefined): ReturnType => {
   const [ensName, setENSName] = useState<string | null>(null)
-  const { library } = useActiveWeb3React()
+  const { library, chainId } = useActiveWeb3React()
 
   useEffect(() => {
     async function resolveENS() {
-      if (address && library && ethers.utils.isAddress(address)) {
+      if (
+        address &&
+        library &&
+        ethers.utils.isAddress(address) &&
+        chainId === ChainId.MAINNET
+      ) {
         const provider = new ethers.providers.Web3Provider(library.provider)
         const ensName = await provider.lookupAddress(address)
+        console.log("ens name on net", ensName)
         if (ensName) setENSName(ensName)
+      } else if (
+        address &&
+        library &&
+        ethers.utils.isAddress(address) &&
```

Comment:
not necessary but putting this logic into a const could clean up a couple lines.